lib/MetadataStore.js

```diff
@@ -231,39 +237,44 @@ MetadataStore.prototype = {
 
         /* 2. inserts the favicon into page_images, handles the special case, in which
          * the favicon already exsits */
-        let faviconBindings = this._getFaviconParameters(metaObject);
-        try {
-          yield this._conn.executeCached(SQL_INSERT_IMAGES, faviconBindings);
-          image_ids.push(yield this._asyncGetLastInsertRowID());
-        } catch (e) {
+        if (metaObject.favicon_url) {
+          let faviconBindings = this._getFaviconParameters(metaObject);
           try {
-            image_ids.push(yield this._asyncGetImageIDByURL(faviconBindings.url));
-          } catch (e) {
-            Cu.reportError("MetadataStore failed to insert favicon: " + e.message);
-            throw e; /* force this transaction to rollback */
-          }
-        }
-
-        /* 3. inserts all the preview images */
-        for (let image of metaObject.images) {
-          let imageBindings = this._getImageParameters(image);
-          try {
-            yield this._conn.executeCached(SQL_INSERT_IMAGES, imageBindings);
+            yield this._conn.executeCached(SQL_INSERT_IMAGES, faviconBindings);
             image_ids.push(yield this._asyncGetLastInsertRowID());
           } catch (e) {
             try {
-              image_ids.push(yield this._asyncGetImageIDByURL(imageBindings.url));
+              image_ids.push(yield this._asyncGetImageIDByURL(faviconBindings.url));
             } catch (e) {
-              Cu.reportError("MetadataStore failed to fetch the id of favicon: " + e.message);
+              Cu.reportError("MetadataStore failed to insert favicon: " + e.message);
               throw e; /* force this transaction to rollback */
             }
           }
         }
 
-        /* 4. inserts relations into the page_metadata_images */
-        for (let image_id of image_ids) {
-          yield this._conn.executeCached(SQL_INSERT_METADATA_IMAGES, [metadata_id, image_id]);
+        /* 3. inserts all the preview images, if they exist */
+        if (metaObject.images) {
+          for (let image of metaObject.images) {
+            let imageBindings = this._getImageParameters(image);
+            try {
+              yield this._conn.executeCached(SQL_INSERT_IMAGES, imageBindings);
+              image_ids.push(yield this._asyncGetLastInsertRowID());
+            } catch (e) {
+              try {
+                image_ids.push(yield this._asyncGetImageIDByURL(imageBindings.url));
+              } catch (e) {
+                Cu.reportError("MetadataStore failed to fetch the id of favicon: " + e.message);
```

Comment:
Oops, found a typo of my previous commit, it should be "MetadataStore failed to fetch the id of **image:** ".
Could you fix this for me, please?